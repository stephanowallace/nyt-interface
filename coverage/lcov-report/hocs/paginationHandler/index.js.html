<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for hocs/paginationHandler/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">hocs/paginationHandler</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>24/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>13/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/22</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">40x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import Constants from 'Constants';
&nbsp;
const mockArticle = {
  byLine: '',
  publishDate: '',
  title: '',
  url: '',
};
&nbsp;
const pageSizeFromStorage = () =&gt; localStorage.getItem('pageSize') || Constants.PAGINATION_SIZE;
const calculateNumberOfPages = contentLength =&gt; Math.ceil(contentLength / pageSizeFromStorage());
&nbsp;
export default (WrappedComponent) =&gt; {
  class PaginationHandler extends Component {
    constructor(props) {
      super(props);
      this.content = [];
      this.state = {
        numberOfPages: 1,
        pageNumber: 1,
        pageSize: pageSizeFromStorage(),
        paginatedContent: Array
          .from(Array(parseInt(pageSizeFromStorage(), 10)))
          .map((_, index) =&gt; ({ ...mockArticle, short_url: index })),
      };
    }
&nbsp;
    getPaginatedContent() {
      const { pageNumber, pageSize } = this.state;
      const start = (pageNumber - 1) * pageSize;
      const end = parseInt(start, 10) + parseInt(pageSize, 10);
      this.setState({ paginatedContent: this.content.slice(start, end) });
    }
&nbsp;
    setContent(contentList) {
      this.content = contentList;
      const pageSize = pageSizeFromStorage();
      this.setState({
        numberOfPages: calculateNumberOfPages(this.content.length),
        pageNumber: 1,
        pageSize,
      }, () =&gt; this.getPaginatedContent());
    }
&nbsp;
    setPageNumber(pageNumber) {
      this.setState({
        pageNumber,
      }, () =&gt; this.getPaginatedContent());
    }
&nbsp;
    setPageSize(pageSize) {
      localStorage.setItem('pageSize', pageSize);
      this.setState({
        numberOfPages: calculateNumberOfPages(this.content.length),
        pageNumber: 1,
        pageSize,
      }, () =&gt; this.getPaginatedContent());
    }
&nbsp;
    render() {
      return (
        &lt;WrappedComponent
          {...this.props}
          pagination={{
            data: { ...this.state },
            setContent: this.setContent.bind(this),
            setPageNumber: this.setPageNumber.bind(this),
            setPageSize: this.setPageSize.bind(this),
          }}
        /&gt;
      );
    }
  }
&nbsp;
  return PaginationHandler;
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 09 2019 20:44:43 GMT-0200 (Brasilia Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
