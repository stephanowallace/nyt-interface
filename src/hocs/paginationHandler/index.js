import React, { Component } from 'react';
import Constants from 'Constants';

const mockArticle = {
  byLine: '',
  publishDate: '',
  title: '',
  url: '',
};

const pageSizeFromStorage = () => localStorage.getItem('pageSize') || Constants.PAGINATION_SIZE;
const calculateNumberOfPages = contentLength => Math.ceil(contentLength / pageSizeFromStorage());

export default (WrappedComponent) => {
  class PaginationHandler extends Component {
    constructor(props) {
      super(props);
      this.content = [];
      this.state = {
        numberOfPages: 1,
        pageNumber: 1,
        pageSize: pageSizeFromStorage(),
        paginatedContent: Array
          .from(Array(parseInt(pageSizeFromStorage(), 10)))
          .map((_, index) => ({ ...mockArticle, short_url: index })),
      };
    }

    getPaginatedContent() {
      const { pageNumber, pageSize } = this.state;
      const start = (pageNumber - 1) * pageSize;
      const end = parseInt(start, 10) + parseInt(pageSize, 10);
      this.setState({ paginatedContent: this.content.slice(start, end) });
    }

    setContent(contentList) {
      this.content = contentList;
      const pageSize = pageSizeFromStorage();
      this.setState({
        numberOfPages: calculateNumberOfPages(this.content.length),
        pageNumber: 1,
        pageSize,
      }, () => this.getPaginatedContent());
    }

    setPageNumber(pageNumber) {
      this.setState({
        pageNumber,
      }, () => this.getPaginatedContent());
    }

    setPageSize(pageSize) {
      localStorage.setItem('pageSize', pageSize);
      this.setState({
        numberOfPages: calculateNumberOfPages(this.content.length),
        pageNumber: 1,
        pageSize,
      }, () => this.getPaginatedContent());
    }

    render() {
      return (
        <WrappedComponent
          {...this.props}
          pagination={{
            data: { ...this.state },
            setContent: this.setContent.bind(this),
            setPageNumber: this.setPageNumber.bind(this),
            setPageSize: this.setPageSize.bind(this),
          }}
        />
      );
    }
  }

  return PaginationHandler;
};
