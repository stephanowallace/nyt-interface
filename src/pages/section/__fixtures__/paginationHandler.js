import articles from 'Fixtures/articles';

module.exports = {
  data: {
    numberOfPages: 1,
    pageNumber: 1,
    pageSize: 10,
    paginatedContent: articles,
  },
  setContent: jest.fn(),
  setPageNumber: jest.fn(),
  setPageSize: jest.fn(),
};
