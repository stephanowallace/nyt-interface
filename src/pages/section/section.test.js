import React from 'react';
import { shallow } from 'enzyme';
import articles from 'Fixtures/articles';
import paginationHandler from './__fixtures__/paginationHandler';
import match from './__fixtures__/params';
import Section from './section';

describe('Section', () => {
  const props = {
    history: { push: jest.fn() },
    match,
    pagination: paginationHandler,
  };

  it('Renders default state', () => {
    const component = shallow(<Section {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('Change page', () => {
    fetch.mockResponseOnce(JSON.stringify({ articles }));
    const component = shallow(<Section {...props} />);
    component.instance().handlePageChange(2);
    expect(paginationHandler.setPageNumber).toHaveBeenCalledWith(2);
  });

  it('Change pagination size', () => {
    fetch.mockResponseOnce(JSON.stringify({ articles }));
    const component = shallow(<Section {...props} />);
    const option = { label: '20', value: '20' };
    component.instance().handlePaginationSizeChange(option);
    expect(paginationHandler.setPageSize).toHaveBeenCalledWith(option.value);
  });

  it('Retrieves new stories if section change', () => {
    fetch.mockResponseOnce(JSON.stringify({ articles }));
    const component = shallow(<Section {...props} />, { lifecycleExperimental: true });
    const instance = component.instance();
    jest.spyOn(instance, 'retrieveTopStoriesFromAPI');
    component.setProps({ match: { params: { section: 'world' } } });
    expect(instance.retrieveTopStoriesFromAPI).toHaveBeenCalled();
  });
});
