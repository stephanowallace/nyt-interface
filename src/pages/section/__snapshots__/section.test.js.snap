// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Renders default state 1`] = `
<CenterContent>
  <PageHeader
    title="science & technology"
  />
  <ArticleList
    articles={
      Array [
        Object {
          "abstract": "The comet, called 46P/Wirtanen, will streak near Earth on Sunday and be visible in some places with the naked eye.",
          "byline": "By CHRISTINA CARON",
          "created_date": "2018-12-15T14:09:49-05:00",
          "des_facet": Array [
            "Comets",
            "46P/Wirtanen",
            "Space and Astronomy",
          ],
          "geo_facet": Array [],
          "item_type": "Article",
          "kicker": "",
          "material_type_facet": "",
          "multimedia": Array [
            Object {
              "caption": "The comet 46P/Wirtanen as seen from France through a telescope this month.",
              "copyright": "Nicolas Biver/Agence France-Presse u2014 Getty Images",
              "format": "Standard Thumbnail",
              "height": 75,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/16/autossell/16xp-comet/16xp-comet-thumbStandard.jpg",
              "width": 75,
            },
            Object {
              "caption": "The comet 46P/Wirtanen as seen from France through a telescope this month.",
              "copyright": "Nicolas Biver/Agence France-Presse u2014 Getty Images",
              "format": "thumbLarge",
              "height": 150,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/16/autossell/16xp-comet/16xp-comet-thumbLarge.jpg",
              "width": 150,
            },
            Object {
              "caption": "The comet 46P/Wirtanen as seen from France through a telescope this month.",
              "copyright": "Nicolas Biver/Agence France-Presse u2014 Getty Images",
              "format": "Normal",
              "height": 127,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/16/autossell/16xp-comet/16xp-comet-articleInline.jpg",
              "width": 190,
            },
            Object {
              "caption": "The comet 46P/Wirtanen as seen from France through a telescope this month.",
              "copyright": "Nicolas Biver/Agence France-Presse u2014 Getty Images",
              "format": "mediumThreeByTwo210",
              "height": 140,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/16/autossell/16xp-comet/merlin_148053384_83232fbc-01e3-4452-aa60-cc085f53bf69-mediumThreeByTwo210.jpg",
              "width": 210,
            },
            Object {
              "caption": "The comet 46P/Wirtanen as seen from France through a telescope this month.",
              "copyright": "Nicolas Biver/Agence France-Presse u2014 Getty Images",
              "format": "superJumbo",
              "height": 1365,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/16/autossell/16xp-comet/16xp-comet-superJumbo.jpg",
              "width": 2048,
            },
          ],
          "org_facet": Array [
            "National Aeronautics and Space Administration",
          ],
          "per_facet": Array [],
          "published_date": "2018-12-15T14:09:49-05:00",
          "section": "Science",
          "short_url": "https://nyti.ms/2En2PqK",
          "subsection": "",
          "title": "A Bright Green u2018Christmas Cometu2019 Will Fly the Closest to Earth in Centuries",
          "updated_date": "2018-12-15T19:28:59-05:00",
          "url": "https://www.nytimes.com/2018/12/15/science/green-comet-46p-wirtanen.html",
        },
        Object {
          "abstract": "The reticulated siren is the largest vertebrate discovered in the United States in decades.",
          "byline": "By ASHER ELBEIN",
          "created_date": "2018-12-14T13:01:09-05:00",
          "des_facet": Array [
            "Salamanders",
            "Amphibians",
            "Biology and Biochemistry",
            "Reptiles",
          ],
          "geo_facet": Array [
            "Alabama",
            "Florida",
          ],
          "item_type": "Article",
          "kicker": "",
          "material_type_facet": "",
          "multimedia": Array [
            Object {
              "caption": "The newest named salamander in the animal kingdom: the reticulated siren.",
              "copyright": "David Steen",
              "format": "Standard Thumbnail",
              "height": 75,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/18/science/18SCI-SALAMANDER1/18SCI-SALAMANDER1-thumbStandard-v2.jpg",
              "width": 75,
            },
            Object {
              "caption": "The newest named salamander in the animal kingdom: the reticulated siren.",
              "copyright": "David Steen",
              "format": "thumbLarge",
              "height": 150,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/18/science/18SCI-SALAMANDER1/18SCI-SALAMANDER1-thumbLarge-v2.jpg",
              "width": 150,
            },
            Object {
              "caption": "The newest named salamander in the animal kingdom: the reticulated siren.",
              "copyright": "David Steen",
              "format": "Normal",
              "height": 127,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/18/science/18SCI-SALAMANDER1/18SCI-SALAMANDER1-articleInline-v2.jpg",
              "width": 190,
            },
            Object {
              "caption": "The newest named salamander in the animal kingdom: the reticulated siren.",
              "copyright": "David Steen",
              "format": "mediumThreeByTwo210",
              "height": 140,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/18/science/18SCI-SALAMANDER1/18SCI-SALAMANDER1-mediumThreeByTwo210-v2.jpg",
              "width": 210,
            },
            Object {
              "caption": "The newest named salamander in the animal kingdom: the reticulated siren.",
              "copyright": "David Steen",
              "format": "superJumbo",
              "height": 1365,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/18/science/18SCI-SALAMANDER1/18SCI-SALAMANDER1-superJumbo-v2.jpg",
              "width": 2048,
            },
          ],
          "org_facet": Array [],
          "per_facet": Array [],
          "published_date": "2018-12-14T13:01:09-05:00",
          "section": "Science",
          "short_url": "https://nyti.ms/2zYXp2d",
          "subsection": "",
          "title": "A Salamander of Legend Emerges From Southern Swamps",
          "updated_date": "2018-12-14T13:01:09-05:00",
          "url": "https://www.nytimes.com/2018/12/14/science/salamander-reticulated-siren.html",
        },
        Object {
          "abstract": "If one duckweed lands where a bird relieves itself, itu2019s capable of eventually creating a dense mat of duckweeds where there were none before.",
          "byline": "By VERONIQUE GREENWOOD",
          "created_date": "2018-12-14T09:32:50-05:00",
          "des_facet": Array [
            "Flowers and Plants",
            "Invasive Species",
            "Weeds",
            "Birds",
            "Ducks",
            "Feces",
            "Wetlands",
          ],
          "geo_facet": Array [
            "Brazil",
            "Great Britain",
            "Spain",
          ],
          "item_type": "Article",
          "kicker": "",
          "material_type_facet": "",
          "multimedia": Array [
            Object {
              "caption": "Duckweed, and a duck, on a pond in Germany.",
              "copyright": "Kay Nietfeld/DPA, via Getty Images",
              "format": "Standard Thumbnail",
              "height": 75,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/15/science/15TB-DUCKWEED1/15TB-DUCKWEED1-thumbStandard.jpg",
              "width": 75,
            },
            Object {
              "caption": "Duckweed, and a duck, on a pond in Germany.",
              "copyright": "Kay Nietfeld/DPA, via Getty Images",
              "format": "thumbLarge",
              "height": 150,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/15/science/15TB-DUCKWEED1/15TB-DUCKWEED1-thumbLarge.jpg",
              "width": 150,
            },
            Object {
              "caption": "Duckweed, and a duck, on a pond in Germany.",
              "copyright": "Kay Nietfeld/DPA, via Getty Images",
              "format": "Normal",
              "height": 126,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/15/science/15TB-DUCKWEED1/15TB-DUCKWEED1-articleInline.jpg",
              "width": 190,
            },
            Object {
              "caption": "Duckweed, and a duck, on a pond in Germany.",
              "copyright": "Kay Nietfeld/DPA, via Getty Images",
              "format": "mediumThreeByTwo210",
              "height": 140,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/15/science/15TB-DUCKWEED1/15TB-DUCKWEED1-mediumThreeByTwo210.jpg",
              "width": 210,
            },
            Object {
              "caption": "Duckweed, and a duck, on a pond in Germany.",
              "copyright": "Kay Nietfeld/DPA, via Getty Images",
              "format": "superJumbo",
              "height": 1364,
              "subtype": "photo",
              "type": "image",
              "url": "https://static01.nyt.com/images/2018/12/15/science/15TB-DUCKWEED1/15TB-DUCKWEED1-superJumbo.jpg",
              "width": 2048,
            },
          ],
          "org_facet": Array [
            "Biology Letters (Journal)",
          ],
          "per_facet": Array [
            "Green, Andy",
            "Silva, Giliandro",
          ],
          "published_date": "2018-12-14T09:32:50-05:00",
          "section": "Science",
          "short_url": "https://nyti.ms/2A0qiLp",
          "subsection": "",
          "title": "One of Natureu2019s Smallest Flowering Plants Can Survive Inside of a Duck",
          "updated_date": "2018-12-14T09:32:51-05:00",
          "url": "https://www.nytimes.com/2018/12/14/science/duckweed-duck-poop.html",
        },
      ]
    }
    loading={true}
    section="sciencetechnology"
  />
  <Pagination
    numberOfPages={1}
    onPageChange={[Function]}
    onPageSizeChange={[Function]}
    selectedPage={1}
    selectedPageSize={
      Object {
        "label": 10,
        "value": 10,
      }
    }
  />
</CenterContent>
`;
